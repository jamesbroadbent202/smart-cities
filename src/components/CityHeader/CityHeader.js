import React from 'react';
import PropTypes from 'prop-types';
import getColorVariant from '../../helpers/getColorVariant';
import Icon from '../Icon/Icon';
import style from './CityHeader.scss';

const shades = [
  '500',
  '600',
  '700',
  '800',
  '900',
];

const CityHeader = (props) => {
  const backgroundPanels = shades.map(shade => (
    <div
      key={shade}
      className={style.backgroundPanel}
      style={{
        background: getColorVariant(props.categoryColorName, shade),
      }}
    />
  ));

  return (
    <div className={style.wrapper}>
      <div className={style.backgroundPanelWrapper}>
        {backgroundPanels}
      </div>

      <div className={style.container}>
        <h1 className={style.title}>
          {props.cityName}
        </h1>

        {props.categoryIconId && (
          <Icon
            className={style.iconWrapper}
            color={getColorVariant(props.categoryColorName, '500')}
            icon={props.categoryIconId}
            size={180}
          />
        )}
      </div>
    </div>
  );
};

CityHeader.propTypes = {
  cityName: PropTypes.string.isRequired,
  categoryIconId: PropTypes.string,
  categoryColorName: PropTypes.string.isRequired,
};

export default CityHeader;
