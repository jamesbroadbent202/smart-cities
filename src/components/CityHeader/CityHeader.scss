@import '~style/variables';

.wrapper {
  position: relative;
  height: 130px;

  @include for-tablet-portrait-up {
    height: 230px;
  }

  @include for-tablet-landscape-up {
    height: 250px;
  }

  @include for-desktop-up {
    height: 290px;
  }
}

.container {
  @include container;

  position: relative; // above the absolute background panels
  display: flex;
  align-items: flex-end;
  padding: 24px 16px;
  height: 100%;

  @include for-tablet-portrait-up {
    padding: 20px;
  }

  @include for-tablet-landscape-up {
    padding: 32px;
  }

  @include for-desktop-up {
    padding: 46px 32px;
  }
}

.backgroundPanelWrapper {
  position: absolute;
  display: flex;
  flex-basis: 0;
  align-items: stretch;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
}

.backgroundPanel {
  flex: auto;

  @include for-tablet-portrait-up {
    &:last-child {
      flex: 2;
    }
  }
}

.title {
  position: relative;
  bottom: -8px; // to bottom-align the text
  line-height: 1.2; // it wraps on small screens
  font-size: 36px;
  font-weight: 300;
  letter-spacing: 0.7px;
  text-shadow: 0 0 4px rgba(0, 0, 0, 0.5);

  @include for-tablet-portrait-up {
    margin-right: 32px;
  }

  @include for-tablet-landscape-up {
    font-size: 48px;
    font-weight: 200;
    bottom: -11px;
  }

  @include for-desktop-up {
    font-size: 72px;
    font-weight: 500;
    bottom: -14px;
  }
}

.iconWrapper {
  margin-left: auto; // push right

  @include for-phone-only {
    display: none;
  }
}
