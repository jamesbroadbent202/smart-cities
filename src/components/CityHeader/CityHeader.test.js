import React from 'react';
import { shallow } from 'enzyme';
import CityHeader from './CityHeader';

jest.mock('../Icon/Icon', () => '../Icon/Icon');

const defaultProps = {
  cityName: 'Greater perth',
  categoryColorName: 'jobs',
  categoryIconId: 'infrastructureGettingToWork',
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityHeader {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
