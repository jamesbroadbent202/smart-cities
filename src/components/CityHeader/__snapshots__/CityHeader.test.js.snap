// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <div className=\\"backgroundPanelWrapper\\">
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
    <div className=\\"backgroundPanel\\" style={{...}} />
  </div>
  <div className=\\"container\\">
    <h1 className=\\"title\\">
      Greater perth
    </h1>
    <../Icon/Icon className=\\"iconWrapper\\" color=\\"JOBS_500\\" icon=\\"infrastructureGettingToWork\\" size={180} />
  </div>
</div>"
`;
